pkg/rpcserver/rpcserver.go

```diff
@@ -278,22 +295,44 @@ func (serv *server) handleRunnerConn(runner *Runner, conn *flatrpc.Conn) error {
 		opts.Features = serv.cfg.Features
 	}
 
-	err := runner.Handshake(conn, opts)
+	info, err := runner.Handshake(conn, opts)
 	if err != nil {
 		log.Logf(1, "%v", err)
 		return err
 	}
 
+	eg, ctx := errgroup.WithContext(baseCtx)
+
+	serv.checkOnce.Do(func() {
+		// Run the machine check.
+		eg.Go(func() error {
+			if err := serv.runCheck(ctx, &info); err != nil {
```

Comment:
Since you used errgroup.WithContext, does it mean that whole checking will be cancelled if the this runner happened to return (a soft) error?

Reply:
Now that these are untied, the checking will be aborted only on hard errors.